sceneview/src/main/java/io/github/sceneview/SceneView.kt

```diff
@@ -497,16 +497,16 @@ open class SceneView @JvmOverloads constructor(
             }
         }
 
-        lifecycle.dispatchEvent<SceneLifecycleObserver> {
-            onFrame(frameTime)
-        }
-        onFrame?.invoke(frameTime)
 
         transformManager.commitLocalTransformTransaction()
+        onFrame?.invoke(frameTime)
 
         // Render the scene, unless the renderer wants to skip the frame.
         if (renderer.beginFrame(swapChain!!, frameTime.nanoseconds)) {
             renderer.render(view)
+            lifecycle.dispatchEvent<SceneLifecycleObserver> {
+                onFrame(frameTime)
+            }
```

Comment:
I would rather not insert any code between the calls to `render` and `endFrame`.

Reply:
Done I've moved it 
From my understanding, onFrame must be called after beginFrame and before endFrame as beginFrame is responsible for drawing the current frame and endFrame finished the current Frame.
The swap chain will only retrieve a surface view or frame only when it is present in the GPU


`if (renderer.beginFrame(swapChain!!, frameTime.nanoseconds)) {
            lifecycle.dispatchEvent<SceneLifecycleObserver> {
                onFrame(frameTime)
            }
            renderer.render(view)
            renderer.endFrame()
        }`

Reply:
I'll need some time to think about that. Unfortunately, I don't find your reasoning very clear because neither of these methods does any drawing on the GPU.

Reply:
I'm sorry 
I might be incorrect in the last line 

_The swap chain will only retrieve a surface view or frame only when it is present in the swapchain not GPU sorry for that_
The renderer.copyFrame function copies the currently rendered View to the indicated SwapChain, using the indicated source and destination rectangle.
copyFrame() should be called after a frame is rendered using render(com.google.android.filament.View) but before endFrame() is called

This is as per the official documentation of Filament 

`copyFrame
public void copyFrame​(@NonNull [SwapChain](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/SwapChain.html) dstSwapChain, @NonNull [Viewport](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/Viewport.html) dst[Viewport](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/Viewport.html), @NonNull Viewport srcViewport, int flags)
Copies the currently rendered [View](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/View.html) to the indicated [SwapChain](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/SwapChain.html), using the indicated source and destination rectangle.
copyFrame() should be called after a frame is rendered using [render(com.google.android.filament.View)](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/Renderer.html?_ijt=j2fp7e4b0l4oskd11kt8f8p82c&_ij_reload=RELOAD_ON_SAVE#render(com.google.android.filament.View)) but before [endFrame()](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/Renderer.html?_ijt=j2fp7e4b0l4oskd11kt8f8p82c&_ij_reload=RELOAD_ON_SAVE#endFrame()) is called.

Parameters:
dstSwapChain - the [SwapChain](http://localhost:63342/SceneView/filament-android-1.31.4-javadoc.jar/com/google/android/filament/SwapChain.html) into which the frame should be copied
dstViewport - the destination rectangle in which to draw the view
srcViewport - the source rectangle to be copied
flags - one or more CopyFrameFlag behavior configuration flags`